docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,109 @@
+## Background
+
+This proposal is to provide CRD level validation for OCS Operator project. Currently, when the user (manually) changes CRD yaml file, the cluster will fail at the creation time if there is an error in the user input. It would have been nice to have a layer of validation when user submit the changes.
```

Comment:
```suggestion
This proposal is to provide validation for OCS Operator CRs. Currently, when the user Creates/Updates the CR, the reconciliation might fail if there is an error in the user input. It will be a better user experience if we validate the request right when user submits the changes and before the CR gets written into etcd.
```